src/CredentialsLoader.php

```diff
@@ -169,4 +169,13 @@ public function updateMetadata(
     $metadata_copy[self::AUTH_METADATA_KEY] = array('Bearer ' . $result['access_token']);
     return $metadata_copy;
   }
+
+  /**
+   * Gets the underlying `OAuth2` instance used to conduct authorization.
+   *
+   * @return OAuth2
+   */
+  public function getAuth() {
```

Comment:
I'm not a huge fan either.

But, the main issue is that if we don't or can't use the middleware classes to help manage our access tokens (we're using SOAPClient), we can't use the getExpiresIn() methods in order to deal with when we need to refresh the access token for the user. The TODOs on the cache usage in the middleware classes also seem to not deal with this properly right now right?

I can always store the access token and calculate this myself, but seems like since the logic is already available, I should be using it from the library.

I also want a common object that abstracts away the underlying flow that's being used, which is why I wanted to use the OAuth2 object. CredentialsLoader could also be that object, except there is an implicit contract for (most of the) cred loader subclasses to impl fetchAuthToken().

I could also expose a method called getExpiresIn() in the cred loaders that I care about that delegates to the OAuth2->getExpiresIn() method as well as another solution. But this wouldn't be defined in the parent or an interface, which makes it a bit risky to assume it's always on a cred loader.